packages/compiler/core/checker.ts

```diff
@@ -2265,15 +2266,19 @@ export function createChecker(program: Program): Checker {
       return;
     }
 
-    if (inheritedPropertyNames?.has(newProp.name)) {
-      program.reportDiagnostic(
-        createDiagnostic({
-          code: "override-property",
-          format: { propName: newProp.name },
-          target: diagnosticTarget ?? newProp,
-        })
-      );
-
+    if (inheritedProperties?.has(newProp.name)) {
+      const inheritedProp = inheritedProperties.get(newProp.name)!;
+      const [isAssignable, _] = isTypeAssignableTo(newProp.type, inheritedProp.type, newProp);
+      const parentIntrinsic = isIntrinsic(program, inheritedProp.type);
```

Comment:
Doesn't this say that we allow any type at all to override parent intrinsic types?

Reply:
Sorry I see, the intention is that we disallow subtyping of non-intrinsic types. Is that what we agreed to in a design discussion?

Reply:
Yes, @timotheeguerin calls it out here: https://github.com/microsoft/cadl/issues/1137

If the derived property isn't assignable to the parent property type or it is but the parent property type isn't intrinsic, it fails.

Reply:
ok seems good for now.